apis/python/src/tiledbsoma/_read_iters.py

```diff
@@ -465,12 +549,46 @@ def _from_table(self, arrow_table: pa.Table) -> pa.SparseCOOTensor:
         return pa.SparseCOOTensor.from_numpy(coo_data, coo_coords, shape=self.shape)
 
 
-def _arrow_table_reader(sr: clib.SOMAArray) -> Iterator[pa.Table]:
-    """Private. Simple Table iterator on any Array"""
-    tbl = sr.read_next()
-    while tbl is not None:
-        yield tbl
-        tbl = sr.read_next()
+class ArrowTableRead:
+    def __init__(
+        self,
+        array: SOMAArray,
+        coords: Union[
+            options.SparseDFCoords, options.SparseNDCoords, options.DenseNDCoords
+        ],
+        column_names: Optional[Sequence[str]],
+        result_order: clib.ResultOrder,
+        value_filter: Optional[str],
+        platform_config: Optional[options.PlatformConfig],
+    ):
+        sr = array._handle._handle
+
+        if platform_config is not None:
+            cfg = sr.context().config()
+            cfg.update(platform_config)
+            ctx = clib.SOMAContext(cfg)
+        else:
+            ctx = sr.context()
+
+        self.mq = clib.ManagedQuery(sr, ctx)
+
+        self.mq.set_layout(result_order)
+
+        if column_names is not None:
+            self.mq.select_columns(list(column_names))
+
+        if value_filter is not None:
+            self.mq.set_condition(QueryCondition(value_filter), sr.schema)
+
+        _util._set_coords(self.mq, sr, coords)
+
+        self.mq.setup_read()
+
+    def __iter__(self) -> Self:
```

Comment:
Iterators don't need an __iter__ method AFAIK.  Iterables do, but iterators do not.  So you should be able to remove this.